wai-aria/scripts/aria-utils.js

```diff
@@ -157,5 +163,35 @@ const AriaUtils = {
   },
 
 
+  /*
+  Tests computed LABEL and ROLE of all elements matching selector using existing
+    verifyLabelsBySelector(), verifyRolesBySelector() functions and passes a test name prefix
+    to ensure uniqueness.
+
+  Ex: <div aria-label="foo" role="button"
+        data-testname="div with role=button is labelled via aria-label"
+        data-expectedlabel="foo"
+        data-expectedrole="button"
+        class="ex">
+
+      AriaUtils.verifyRoleAndLabelBySelector(".ex")
+
+  */
+  verifyRoleAndLabelBySelector: function(selector) {
+    let labelTestNamePrefix = "(Label Test) ";
+    let roleTestNamePrefix = "(Role Test) ";
+
+    const els = document.querySelectorAll(selector);
+    if (!els.length) {
+      throw `Selector passed in verifyRoleAndLabelBySelector("${selector}") should match at least one element.`;
+    }
+    for (const el of els) {
+      el.classList.add("ex-label-only");
+      el.classList.add("ex-role-only");
+    }
+
+    this.verifyLabelsBySelector(".ex-label-only", labelTestNamePrefix);
+    this.verifyRolesBySelector(".ex-role-only", roleTestNamePrefix);
```

Comment:
Interesting... I would not have thought to do it this way. I assumed you'd have added role and label promise test blocks in the `for` loop…

…but I think this is also fine.

Reply:
Right, it's desirable to have each test be grouped so upon execution, the results are displayed in the dashboard as: 

- role: test a
- label: test a
- role: test b
- label: test b

However, the `verifyRolesBySelector()` and `verifyLabelsBySelector()` logic would need further changes so it doesn't iterate over all elements with that selector. The other (undesirable) alternative would be duplicating the code in verifyRoleAndLabelBySelector() but this introduces non-modular code (separate functions that do the same thing in slightly different ways).

I opted to reuse the above functions but it means I have to call them separately.